src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -0,0 +1,278 @@
+                       "Specific heat capacity must be positive in the linear isotropic conductor material model.");
+  }
+
+  /**
+   * @brief Function defining the thermal flux (constitutive response)
+   *
+   * @tparam T1 type of the temperature (e.g. tensor or dual type)
+   * @tparam T2 type of the temperature gradient (e.g. tensor or dual type)
+   * @param temperature_gradient Gradient of the temperature (du_dx)
+   * @return The thermal flux of the material model
+   */
+  template <typename T1, typename T2>
+  SERAC_HOST_DEVICE T2 operator()(T1& /* temperature */, T2& temperature_gradient) const
+  {
+    return -1.0 * conductivity_ * temperature_gradient;
+  }
+
+  /**
+   * @brief The density (mass per volume) of the material model
+   *
+   * @tparam dim The dimension of the problem
+   * @return The density
+   */
+  template <int dim>
+  SERAC_HOST_DEVICE double density(tensor<double, dim>& /* x */) const
+  {
+    return density_;
+  }
+
+  /**
+   * @brief The specific heat capacity (heat capacity per unit mass) of the material model
+   *
+   * @tparam dim The dimension of the problem
+   * @tparam T Type of the temperature variable
+   */
+  template <typename T, int dim>
+  SERAC_HOST_DEVICE double specificHeatCapacity(tensor<double, dim>& /* x */, T& /* temperature */) const
+  {
+    return specific_heat_capacity_;
+  }
+
+private:
+  /// Density
+  double density_ = 1.0;
+
+  /// Specific heat capacity
+  double specific_heat_capacity_ = 1.0;
+
+  /// Constant isotropic thermal conductivity
+  double conductivity_ = 1.0;
+};
+
+/**
+ * @brief Linear anisotropic thermal material model
+ *
+ * @tparam dim Spatial dimension
+ */
+template <int dim>
+class LinearConductor {
+public:
+  /**
+   * @brief Construct a new Linear Isotropic Conductor object
+   *
+   * @param density Density of the material (mass/volume)
+   * @param specific_heat_capacity Specific heat capacity of the material (energy / (mass * temp))
+   * @param conductivity Thermal conductivity of the material (power / (length * temp))
+   */
+  LinearConductor(double density = 1.0, double specific_heat_capacity = 1.0,
+                  tensor<double, dim, dim> conductivity = {{{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}}})
+      : density_(density), specific_heat_capacity_(specific_heat_capacity), conductivity_(conductivity)
+  {
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(specific_heat_capacity_ < 0.0,
+                       "Specific heat capacity must be positive in the linear conductor material model.");
+
+    // Check that the conductivity tensor is symmetric
+    for (int i = 0; i < dim; ++i) {
+      for (int j = i + 1; j < dim; ++j) {
+        SLIC_ERROR_ROOT_IF(std::abs(conductivity_(i, j) - conductivity_(j, i)) > 1.0e-7,
+                           "Conductivity tensor must be symmetric for the linear conductor material model.");
+      }
+    }
```

Comment:
rather than implement these checks in the ctor (and later duplicate them in sister material models), we could expand the tensor library with an appropriately-named function:

```cpp
template < int n >
bool is_symmetric(tensor< double, n, n > A, double tolerance = 1.0e-8) {
  for (int i = 0; i < n; ++i) {
    for (int j = i + 1; j < n; ++j) {
      if (std::abs(A(i, j) - A(j, i)) > tolerance) { return false; };
    }
  }
  return true;
}
```

and that would let the code document itself, without relying on an inline comment to explain what is going on:

```cpp
SLIC_ERROR_ROOT_IF(!is_symmetric(conductivity_), "error message");
```

Reply:
(minor) also, does SLIC have an assertion (i.e. check that the predicate is true, rather than false)? I think it is more natural to say these sort of checks as `assert(is_symmetric(A))` than the double-negative of `error_if(!is_symmetric(A))`

Reply:
@white238 , are there any SLIC calls (other than assert) that do this? I would like to keep these checks in release builds.

Reply:
There is not.

Reply:
To expand: 

We could add one on our side though via a macro that just negates the expression.  Though I personally prefer `SLIC_ERROR_IF(!is_symmetric(A))` but I admit this is just a personal preference.

Asserts (throws an error)/checks (throws a warning) are both in SLIC but are turned into noops when Axom is not built debug.